src/skills-builder/skills-builder-steps/view-results/ViewResults.jsx

```diff
@@ -38,6 +38,7 @@ const ViewResults = () => {
 
   useEffect(() => {
     const getAllRecommendations = async () => {
+      setIsLoading(true);
```

Comment:
Two questions/thoughts: 
1. If `isLoading` is set to true on declaration on line 32, what is the reason for setting it here? I'm assuming you had to because it wasn't working before, likely due to follow-up renders after the page initially rendered? 
2. `getAllRecommendations` _probably_ should be extracted outside of this `useEffect`? It'd be much easier to test, I imagine.

Reply:
1. Excellent question! For the single page version, we are changing the values inside `careerInterests` whenever the user makes a selection in the dropdown. This will trigger this `useEffect` to fire and re-render our new recommendations. Now a loading indicator will appear in between selections on the single page version. This has no implications on the two page version. For that version, all recommendations are loaded at once and then the selectable box radio buttons flip through each one. The `careerInterests` are only set at the beginning in the first step/page.

2. `getAllRecommendations` is wrapping `getRecommendations`, which is imported from the `service.js` file. Initially, `getRecommendations` was not extracted and was all just inside the `useEffect`.  Now `getAllRecommendations` simply calls that request and then sets a handful of local state variables. I don't have it outside the `useEffect` because I understood it as best practice to declare the function inside the hook as long as it's not called anywhere else. If we put it outside the `useEffect`, we'll have to list it as a dependency of the hook and then also memoize it with `useCallback`

Reply:
Sorry about the delay in response, but this all looks good and thanks so much for explaining the thought process! TIL about useEffect best practice